cardano-c/src/block.rs

```diff
@@ -0,0 +1,69 @@
+use cardano::block;
+use std::ptr;
+use types::{BlockPtr, CardanoResult, SignedTransactionPtr};
+
+#[no_mangle]
+pub extern "C" fn cardano_raw_block_decode(
+    bytes: *const u8,
+    size: usize,
+    out: *mut BlockPtr,
+) -> CardanoResult {
+    let slice = unsafe { std::slice::from_raw_parts(bytes, size) };
+    let raw_block = block::block::RawBlock::from_dat(slice.to_vec());
+    let block = match raw_block.decode() {
+        Ok(b) => b,
+        Err(_) => return CardanoResult::failure(),
+    };
+    let pointer = Box::into_raw(Box::new(block));
```

Comment:
Why not?

```suggestion
    let pointer = Box::new(block).into_raw();
```

Reply:
I don't think I can do that, the compiler tells me that `into_raw` is an associated function, not a method?

And according to the Rust documentation:
> Note: this is an associated function, which means that you have to call it as Box::into_raw(b) instead of b.into_raw(). This is so that there is no conflict with a method on the inner type.